### PR TITLE
Sound

### PR DESCRIPTION
**Pull request by ArchDemons**
Test new sound system ^_^.  Need some magic with gui sounds. But not in this PR

### CODE REVIEW

File: src/toniarts/openkeeper/game/sound/SoundCategory.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * Get SdtFile
+     *
+     * @param archiveFilename name in BankMapFile
+     * @return SdtFile
+     */
+    @Nullable
+    public static SdtFile getSdtFile(String archiveFilename) {
+
+        for (String part : new String[] {"HD.sdt", "HD.SDT", "HW.sdt", "HW.SDT"}) {
```
Comment (ufdada) [inline src/toniarts/openkeeper/game/sound/SoundCategory.java]: Shouldn't `ConversionUtils.getRealFileName` be better instead of duplicate entries?

Comment (tonihele) [inline src/toniarts/openkeeper/game/sound/SoundCategory.java]: I never understood the difference between HD & HW... Now we prefer HD files, is that just a guess?

**Comment by tonihele** [comment]
Oh nice, you can solve conflicts in Github straight...

**Comment by tonihele** [comment]
Hmm, there is a lot of great work there, but what is the rationale on extracting the HDs and HWs? There are now 2000 duplicate sounds. Not really an issue on today's super computers, but doesn't feel so elegant? Maybe better for debugging.

**Comment by ArchDemons** [comment]
when we decide what better HD or HW, we can delete unnecessary files

**Comment by tonihele** [comment]
But changing it with this kind of system requires some kind of redoing? Or did it already have a switch somewhere? I'm just afraid the filenames get more hard coded with the path.

**Comment by ArchDemons** [comment]
It will not be a problem to redo it. Conversely, now HW and HD create more problems

**Comment by ArchDemons** [comment]
I heard the difference. HD - clear sound, HW  - as with echo (the environment) in an empty room or in a dungeon.

**Comment by tonihele** [comment]
Hmm, maybe there is a flag or something to use environmental or pure. Was it that there was more HWs, or which way around it was?

If we find no logic which to play. We could use HD always, and add an option for environmental sound which would mean those OpenAL environmental audio, might be better quality than pre-made low quality files although of course not the same.  And this is just an assumption, I have no idea about sound. And I listen to way too much heavy metal.

**Pull request merged**